dev

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+## dev is a script to help us get started performing typical task during local
+## development without needing to learn everything at once
+##
+## - init
+## - start-k8s
+## - upgrade
+## - lint-and-validate
+## - test
+##
+
+## if the script is sourced
+if [ "${BASH_SOURCE[0]}" != "${0}" ]; then
+    if ! [ "$1" = "init" ]; then
+        echo "Only source the init command, run your command without a leading dot!"
+    else
+        if [ "$1" = "init" ]; then
+            . ./ci/common
+            pip3 install -r dev-requirements.txt
```

Comment:
i use minikube and for BinderHub we will (have to) keep using that setup for a while. No strong feelings either way. Maybe a downside of a new kind of VM is that I will end up having minishift, minikube and vagrant because different projects use different setups.

Reply:
An alternative to directly calling `pip`/`pip3` is to use `python -m pip` instead of `pip` as command. I think this has a very high chance of invoking the Python executable that the user actually wants us to use.

Python 2 will be sunsetted in a few months so I think we can stop catering to people who have it installed as their main Python.

Just looked if I have `pip3` on my system. The answer is yes but it is connected to the system python (despite having a conda env activated). So at least for me `pip3 install <stuff>` would lead to "disaster".

Reply:
I'm leaning towards making checks to ensure we are in the VM as can be confirmed by the fact that we are running as the user named vagrant for example.

In the current image ubuntu bionic, there will only be a single Python, and its the system python, and we don't introduce any virtual envs etc because the whole VM is our virtual environment so to say. It reduce the complexity quite significantly.

Reply:
Is `dev` meant to be run in the VM or by the user? We should make a very clear and strong distinction between commands that are meant to be run in the VM vs commands directly exposed to the user in their environment.

Reply:
@minrk yepp! I wholeheartedly agree that we should make it very hard to do it the wrong way.

I opened #1432 to discuss the use of a VM. I figured this discussion could be an important one to reference later and that it would easier to find from that issue where it can also become dedicated.